File: app/src/main/java/no/joharei/flixr/login/LoginApi.java
package no.joharei.flixr.login;

import javax.inject.Inject;

import no.joharei.flixr.api.FlickrApiContainer;
import no.joharei.flixr.login.models.Login;
import no.joharei.flixr.tools.ObservableCache;
import rx.Observable;

public class LoginApi {
    private FlickrApiContainer flickrApiContainer;
    private ObservableCache observableCache;

    @Inject
    LoginApi(FlickrApiContainer flickrApiContainer, ObservableCache observableCache) {
        this.flickrApiContainer = flickrApiContainer;
        this.observableCache = observableCache;
    }

    Observable<Login> getUserDetails() {
        return flickrApiContainer.getUserDetails();
    }

    public void clearCache() {
        observableCache.clearCache();
    }
}

File: app/src/main/java/no/joharei/flixr/login/LoginApi.kt
package no.joharei.flixr.login

import io.reactivex.Observable
import no.joharei.flixr.login.models.User
import no.joharei.flixr.network.FlickrApiContainer
import no.joharei.flixr.tools.ObservableCache
import javax.inject.Inject

class LoginApi
@Inject
constructor(private val flickrApiContainer: FlickrApiContainer, private val observableCache: ObservableCache) {

    fun fetchUserDetails(): Observable<User> {
        return flickrApiContainer.getUserDetails()
    }

    fun clearCache() {
        observableCache.clearCache()
    }
}
